LEGO1/lego/legoomni/src/worlds/legoact2.cpp

```diff
@@ -938,169 +939,169 @@ MxResult LegoAct2::BadEnding()
+			StartAction(Act2mainScript::c_snsx10ni_RunAnim, TRUE, FALSE, NULL, NULL, NULL);
 		}
 		else {
-			FUN_10052560(Act2mainScript::c_snsx11ni_RunAnim, TRUE, FALSE, NULL, NULL, NULL);
+			StartAction(Act2mainScript::c_snsx11ni_RunAnim, TRUE, FALSE, NULL, NULL, NULL);
 		}
 		break;
 	case 38:
 	case 42:
 		if (randN == 1) {
-			FUN_10052560(Act2mainScript::c_snsx03ma_RunAnim, TRUE, FALSE, NULL, NULL, NULL);
+			StartAction(Act2mainScript::c_snsx03ma_RunAnim, TRUE, FALSE, NULL, NULL, NULL);
 		}
 		else {
-			FUN_10052560(Act2mainScript::c_snsx04ma_RunAnim, TRUE, FALSE, NULL, NULL, NULL);
+			StartAction(Act2mainScript::c_snsx04ma_RunAnim, TRUE, FALSE, NULL, NULL, NULL);
 		}
 		break;
 	}
 }
 
 // FUNCTION: LEGO1 0x100521f0
 // FUNCTION: BETA10 0x100142f1
-void LegoAct2::FUN_100521f0(MxS32 p_param1)
+void LegoAct2::CheckBricksterIsLoose(MxS32 p_pathData)
 {
 	Act2mainScript::Script objectId = (Act2mainScript::Script) 0;
-	Mx3DPointFloat vec;
+	Mx3DPointFloat position;
 
-	switch (p_param1) {
+	switch (p_pathData) {
 	case 0x02: {
-		vec = Mx3DPointFloat(-9.1f, 0.0f, -16.5f);
+		position = Mx3DPointFloat(-9.1f, 0.0f, -16.5f);
 		VariableTable()->SetVariable("ACTOR_01", "bd");
 		objectId = Act2mainScript::c_tns030bd_RunAnim;
 		break;
 	}
 	case 0x2a: {
-		vec = Mx3DPointFloat(-9.67f, 0.0f, -44.3f);
+		position = Mx3DPointFloat(-9.67f, 0.0f, -44.3f);
 		VariableTable()->SetVariable("ACTOR_01", "rd");
 		objectId = Act2mainScript::c_tns030rd_RunAnim;
 		break;
 	}
 	case 0x133: {
-		vec = Mx3DPointFloat(25.75f, 0.0f, -13.0f);
+		position = Mx3DPointFloat(25.75f, 0.0f, -13.0f);
 		VariableTable()->SetVariable("ACTOR_01", "pg");
 		objectId = Act2mainScript::c_tns030pg_RunAnim;
 		break;
 	}
 	case 0x134: {
-		vec = Mx3DPointFloat(43.63f, 0.0f, -46.33f);
+		position = Mx3DPointFloat(43.63f, 0.0f, -46.33f);
 		VariableTable()->SetVariable("ACTOR_01", "sy");
 		objectId = Act2mainScript::c_tns030sy_RunAnim;
 		break;
 	}
 	case 0x135: {
-		vec = Mx3DPointFloat(50.0f, 0.0f, -34.6f);
+		position = Mx3DPointFloat(50.0f, 0.0f, -34.6f);
 		VariableTable()->SetVariable("ACTOR_01", "rd");
 		objectId = Act2mainScript::c_tns030rd_RunAnim;
 		break;
 	}
 	case 0x138: {
-		vec = Mx3DPointFloat(-41.15f, 4.0f, 31.0f);
+		position = Mx3DPointFloat(-41.15f, 4.0f, 31.0f);
 		VariableTable()->SetVariable("ACTOR_01", "sy");
 		objectId = Act2mainScript::c_tns030sy_RunAnim;
 		break;
 	}
 	}
 
 	if (objectId != (Act2mainScript::Script) 0) {
-		Mx3DPointFloat local30(vec);
-		Mx3DPointFloat position(m_pepper->GetWorldPosition());
-		local30 -= position;
-		Mx3DPointFloat local44 = local30;
-		local30.Unitize();
-		FUN_10052560(objectId, TRUE, TRUE, &vec, &local30, NULL);
+		Mx3DPointFloat lookingAtPepper(position);
+		Mx3DPointFloat positionPepper(m_pepper->GetWorldPosition());
+		lookingAtPepper -= positionPepper;
+		Mx3DPointFloat local44 = lookingAtPepper;
+		lookingAtPepper.Unitize();
+		StartAction(objectId, TRUE, TRUE, &positionPepper, &lookingAtPepper, NULL);
```

Comment:
This is still incorrect. You are introducing a bug here. Please carefully look at which arguments / variables you are passing compared to the original version. Double check everything else.

Reply:
It should now be correct with starting the action at the actor's position which isn't being shadowed the original variable name.